src/Qowaiv/Uuid.cs

```diff
@@ -7,436 +7,435 @@
+        public static implicit operator Uuid(Guid val) => new Uuid(val);
+
+        #endregion
+
+        #region Factory methods
+
+        /// <summary>Initializes a new instance of a UUID.</summary>
+        public static Uuid NewUuid()
+        {
+            return new Uuid(Guid.NewGuid());
+        }
+
+        /// <summary>Converts the string to a UUID.</summary>
+        /// <param name="s">
+        /// A string containing a GUID to convert.
+        /// </param>
+        /// <returns>
+        /// A UUID.
+        /// </returns>
+        /// <exception cref="FormatException">
+        /// s is not in the correct format.
+        /// </exception>
+        public static Uuid Parse(string s)
+        {
+            Uuid val;
+            if (Uuid.TryParse(s, out val))
+            {
+                return val;
+            }
+            throw new FormatException(QowaivMessages.FormatExceptionQGuid);
+        }
+
+        /// <summary>Converts the string to a UUID.
+        /// A return value indicates whether the conversion succeeded.
+        /// </summary>
+        /// <param name="s">
+        /// A string containing a UUID to convert.
+        /// </param>
+        /// <returns>
+        /// The GUID if the string was converted successfully, otherwise QGuid.Empty.
+        /// </returns>
+        public static Uuid TryParse(string s)
+        {
+            Uuid val;
+            if (Uuid.TryParse(s, out val))
+            {
+                return val;
+            }
+            return Uuid.Empty;
+        }
+
+        /// <summary>Converts the string to a UUID.
+        /// A return value indicates whether the conversion succeeded.
+        /// </summary>
+        /// <param name="s">
+        /// A string containing a UUID to convert.
+        /// </param>
+        /// <param name="result">
+        /// The result of the parsing.
+        /// </param>
+        /// <returns>
+        /// True if the string was converted successfully, otherwise false.
+        /// </returns>
+        public static bool TryParse(string s, out Uuid result)
+        {
+            result = Uuid.Empty;
+            if (string.IsNullOrEmpty(s))
+            {
+                return true;
+            }
+
+            if (Pattern.IsMatch(s))
+            {
+                var bytes = Convert.FromBase64String(s.Replace('-', '+').Replace('_', '/').Substring(0, 22) + "==");
+                result = new Uuid(new Guid(bytes));
+                return true;
+            }
+
+            Guid id;
+            if (Guid.TryParse(s, out id))
+            {
+                result = new Uuid(id);
+                return true;
+            }
+            return false;
+        }
+
+        /// <summary>Generates an <see cref="Uuid"/> applying a <see cref="MD5"/> hash on the data.</summary>
+        public static Uuid GenerateWithMD5(byte[] data)
+        {
+            Guard.NotNull(data, nameof(data));
+            var hash = MD5.Create().ComputeHash(data);
```

Comment:
Shouldn't the instance of MD5 be disposed?

Reply:
It should. 👍